flow/core/kernel/vehicle/traci.py

```diff
@@ -554,6 +585,12 @@ def get_speed(self, veh_id, error=-1001):
         if isinstance(veh_id, (list, np.ndarray)):
             return [self.get_speed(vehID, error) for vehID in veh_id]
         return self.__sumo_obs.get(veh_id, {}).get(tc.VAR_SPEED, error)
+    
+    def get_power(self, veh_id, error=-1001):
+        """See parent class."""
+        if isinstance(veh_id, (list, np.ndarray)):
+            return [self.get_power(vehID, error) for vehID in veh_id]
+        return self.__vehicles.get(veh_id, {}).get("energy_model", error)
```

Comment:
I think we need to create this as an abstractmethod in the base vehicle class with a proper docstring. Let's rename this to `get_energy_model` to be a bit more descriptive.

Also, I think if this returns `error`, then later it will break if we try to call `error.get_instantaneous_power()`

Reply:
+1 to all of the above